### DISCUSSION TITLE
Information Required

### DISCUSSION THREAD

**Discussion by biplab123321** [aws/aws-connected-device-framework#82] [General]
I was looking into this repository I would like to understand what is the use of the client folder (aws-connected-device-framework/tree/main/source/packages/libraries)/clients/)?  How is the folder is generated? please share any documentation available.

Regards thanks.

**Comment by aaronatbissell** [comment]
Hi @biplab123321,

The clients are used as an interface to some of the services found in the services folders. For example, if you are using the `assetlibrary` [service](https://github.com/aws/aws-connected-device-framework/tree/main/source/packages/services/assetlibrary), the way to interface with `assetlibrary` is through the `assetlibrary-client`. So if you are using a javascript project, you can include the `@cdf/assetlibrary-client` package into your facade code as a way of interfacing with the `assetlibrary` service.

This is the same for all the other major components of the system (commands, notifications, provisioning, etc)

Hope this helps!

**Reply by biplab123321** [reply]
Thank you, do we have any tutorial for the CDF available? I was searching for more resources over the internet and AWS workshop and documents but didn't find any.
It will be great if you could point me to some.

Regards
